mlflow/langchain/chat_agent_langgraph.py

```diff
@@ -298,10 +304,27 @@ def predict_stream(
         context: Optional[ChatContext] = None,
         custom_inputs: Optional[dict[str, Any]] = None,
     ) -> Generator[ChatAgentChunk, None, None]:
-        for event in self.agent.stream(
-            {"messages": self._convert_messages_to_dict(messages)}, stream_mode="updates"
-        ):
+        request = {
+            "messages": self._convert_messages_to_dict(messages),
+            **({"custom_inputs": custom_inputs} if custom_inputs else {}),
+            **({"context": context.model_dump_compat()} if context else {}),
+        }
+
+        last_message = None
+        last_custom_outputs = None
+
+        for event in self.agent.stream(request, stream_mode="updates"):
             for node_data in event.values():
                 if not node_data:
                     continue
-                yield from (ChatAgentChunk(**{"delta": msg}) for msg in node_data["messages"])
+                messages = node_data.get("messages", [])
+                custom_outputs = node_data.get("custom_outputs")
+
+                for message in messages:
+                    if last_message:
+                        yield ChatAgentChunk(delta=last_message)
```

Comment:
This means we only yield the current message once the next message has been fully generated right?